packages/fake-plugin/lib/plugin.js

```diff
@@ -4,12 +4,22 @@
 import {BasePlugin} from 'appium/plugin';
 import B from 'bluebird';
 
+/**
+ * An example plugin for Appium that demonstrate the implementations.
+ *
+ * @extends {BasePlugin}
```

Comment:
`@extends` is _only_ necessary if the class that you're extending expects a type argument (like `BaseDriver`).

However, we _do_ want to ensure that `FakeDriver` implements `Plugin`, because for some reason we have to be explicit about it.

```suggestion
 * @implements {Plugin}
```

You'll need a `@typedef` for `Plugin` referencing `import('@appium/types').Plugin` somewhere, because we cannot use `import(..)` in an `@implements` tag (probably a bug).